### PR TITLE
(6/10) RUM-2895 Create the ComposeWireframeMapper

### PR DESCRIPTION
**Pull request by xgouchet**
_This PR has been created automatically by splitting a large branch_

RUM-2895 Create the ComposeWireframeMapper

### CODE REVIEW

File: features/dd-sdk-android-session-replay-compose/api/apiSurface [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class com.datadog.android.sessionreplay.compose.mappers.ComposeWireframeMapper : com.datadog.android.sessionreplay.internal.recorder.mapper.BaseWireframeMapper<androidx.compose.ui.platform.ComposeView, com.datadog.android.sessionreplay.model.MobileSegment.Wireframe>
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/api/apiSurface:1]: should this one be public?

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/mappers/ComposeWireframeMapper.kt [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        // "TabRow" : holds the tab row bg color
+        //  "Tab": holds selected tab info
+    )
+
+    override fun map(
+        view: ComposeView,
+        mappingContext: MappingContext,
+        asyncJobStatusCallback: AsyncJobStatusCallback
+    ): List<MobileSegment.Wireframe> {
+        val density = mappingContext.systemInformation.screenDensity.let { if (it == 0.0f) 1.0f else it }
```
Comment (mariusc83) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/mappers/ComposeWireframeMapper.kt:46]: when this is happening ? Do we have to apply this workaround also for the other mappers ?

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/ComposeViewUtils.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            .firstOrNull()
+    } else {
+        null
+    }
+}
+
+private val wrappedCompositionTagKey = androidx.compose.ui.R.id.wrapped_composition_tag
+
+private fun getViewComposer(view: View): Composer? {
+    var composition = (view.getTag(wrappedCompositionTagKey) as? Composition) ?: return null
```
Comment (mariusc83) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/ComposeViewUtils.kt:34]: is this something that we add at compile time ?

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/mappers/ComposeWireframeMapper.kt [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        val startNs = SystemClock.elapsedRealtimeNanos()
+        createComposerWireframes(composer, wireframes, UiContext(null, density))
+        val stopNs = SystemClock.elapsedRealtimeNanos()
+
+        val totalQuery = ComposeContext.contextCache.hitCount() + ComposeContext.contextCache.missCount()
+        val hitRatio = ComposeContext.contextCache.hitCount() * 100.0 / totalQuery
+        val missRatio = ComposeContext.contextCache.missCount() * 100.0 / totalQuery
+
+        // During development, let's keep this telemetry to ensure we keep track of our performance
+        (Datadog.getInstance() as? FeatureSdkCore)?.internalLogger?.log(
```
Comment (mariusc83) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/mappers/ComposeWireframeMapper.kt:88]: Nice one

**Review by jonathanmos — APPROVED** [review]

File: features/dd-sdk-android-session-replay-compose/api/apiSurface (hunk 1 above)
Comment (xgouchet) [reply features/dd-sdk-android-session-replay-compose/api/apiSurface:1]: Good point, it doesn't have to indeed

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/mappers/ComposeWireframeMapper.kt (hunk 2 above)
Comment (xgouchet) [reply features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/mappers/ComposeWireframeMapper.kt:46]: Indeed, it doesn't happen, I think during some of my test I was trying some stuff and it remained.

**Review by 0xnm — APPROVED** [review]

**Pull request merged**